### ISSUE TITLE
[Suggestion] Can the player inherit their velocity from child entities instead of always inheriting from the main entity?

### ISSUE THREAD

**Issue by daOyster** [tsunamayo/Starship-EVO#5041]
I'd like to suggest a change to how the game handles the player inheriting the velocity and inertia of ships when inside of a ship gravity field. Currently the game always inherits the velocity of the base grid when inside a ships gravity field. This causes the player to slide off a horizontal sliding platform if you try to make one, even if it has a gravity block attached to the child entity. What I would like to suggest is to have the player inherit the velocity of the inner most child entity that has a gravity field the player is contained in. 

Take this example scenario in this photo:
![image](https://user-images.githubusercontent.com/1314048/218292192-cb3e5008-6d5d-46df-b7a7-c011cdecae63.png)
In this situation where there are two overlapping gravity fields on the base grid and on the end of a piston, it would be nice if the player would inherit the velocity of the child entity they are standing on/in instead of the base grid. This would allow for more functional sideways elevators and trams on things like stations and large ships. Currently if you try this the floor will slide right out from under you.

My experiments with the conveyor belts suggest to me it's possible for the game to handle a different physics parent when you are touching them while on a moving child entity, but it would be nice if you didn't need to use workarounds to make a platform the moves sideways. Thanks for taking the time to read this!

**Comment by piotmetans** [comment]
I've been wanting to use this tram system without having to sit down. This is a great solution.
https://user-images.githubusercontent.com/660015/218293498-79d79213-1563-4d8f-ac91-21e231c3f1e4.mp4

**Comment by Dwarf-LordPangolin** [comment]
Agreed. This would be a very helpful feature to have. If possible, this would be nice to have for entity on entity contact as well, i.e. ships resting on moving objects moving with them.

**Comment by tsunamayo** [comment]
Hi @daOyster really sorry I am only reading this now, I was crunching on the planet update. So you mean the rule would be the smallest entity gets picked? Or the lowest children entity? It is a bit tricky. 
I could also get a priority setting on the gravity unit.
@piotmetans @Dwarf-LordPangolin If you have a take on that.
By the way the conveyor works very differently it is only adding a linear speed, it is not a parenting, but I could also try to integrate that with CE speed to.
Cheers
